source/decode.c

```diff
@@ -497,17 +616,22 @@ static int s_linestate_response(struct aws_http_decoder *decoder, struct aws_byt
 
     struct aws_byte_cursor version_expected = aws_http_version_to_str(AWS_HTTP_VERSION_1_1);
     if (!aws_byte_cursor_eq(&version, &version_expected)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_STREAM, "id=%p: Incoming response uses unsupported HTTP version.", decoder->logging_id);
         return aws_raise_error(AWS_ERROR_HTTP_PARSE);
     }
 
-    int64_t code_val;
-    int err = s_read_int64(code, &code_val);
+    size_t code_val;
+    err = s_read_size(code, &code_val, 10);
     if (err) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_STREAM, "id=%p: Failed to parse status code from incoming response.", decoder->logging_id);
         return AWS_OP_ERR;
     }
 
     /* Status-code is a 3-digit integer. RFC7230 section 3.1.2 */
-    if (code.len != 3 || code_val < 0 || code_val > 999) {
+    if (code.len != 3 || code_val > 999) {
+        AWS_LOGF_ERROR(AWS_LS_HTTP_STREAM, "id=%p: Incoming response has invalid status code", decoder->logging_id);
```

Comment:
Worth logging some form of what that code was?

Reply:
did